.github/workflows/ci.yaml

```
@@ -0,0 +1,74 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - staging
+      - trying
+
+env:
+  RUSTFLAGS: -Dwarnings
+
+jobs:
+  build_and_test:
+    name: Build and test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [nightly]
```

We have async/await on stable, so why not test on it? 😜 
```suggestion
        rust: [stable, nightly]
```

I wonder if we have a way to make nightly tests not affect the CI status.